imap-types/src/auth.rs

```diff
@@ -56,6 +56,21 @@ pub enum AuthMechanism<'a> {
     /// + draft-murchison-sasl-login-00: The LOGIN SASL Mechanism
     Login,
 
+    /// Standard OAuth 2.0 mechanism.
```

Comment:
How about ...

```suggestion
    /// OAuth 2.0 bearer token mechanism.
```

...?